### ISSUE TITLE
Could not set localStorage error

### ISSUE THREAD

**Issue by lfkpoa** [gzuidhof/starboard-observable#3]
Hi.
Congratulations!
Starboard notebook is awesome and this plugin is really great.
ObservableHQ is changing their pricing plans and now we will need something like starboard.

I'm loading the startboard-observable plugin in the first cell and I'm getting a warning for each cell of type observable:
"Could not set localStorage (key = starboard_notebook_text_editor_preference). "
When I open https://starboard.gg/gz/open-source-observablehq-nfwK2VA this also happens.

Best regards.

**Comment by gzuidhof** [comment]
Hi lfkpoa,

I must say that this project (Observable in Starboard) never really got adoption, which is a shame as it's an open alternative. I imagine that some more dependencies should be updated for this to work again as expected.

This local storage issue is likely due to storage permissions, which Chrome has made more strict at some point. The fix would be to go in the source and make it optional (and ask for permission - I am sure the user is happy to give permission). 

As for Starboard itself gaining adoption, I believe that 80% of the work has been done, but there last 20% of work always takes 80% of the effort. The main way that I would love to see it work is like `diagrams.net`, where there is a frontend app which you link to your storage of choice (e.g. Google Drive) - that way it's free forever, self-hostable and sharable without the lock-in that is required for Observable as a company.

But I'm digressing, I'm happy to accept a PR that incorporates a fix for this. The `try/catch` used to do the job https://github.com/gzuidhof/starboard-notebook/blob/80123a45d37bca591f5d3e04e098ef2e1234dd6a/packages/starboard-notebook/src/components/textEditor.ts#L39-L46  and https://github.com/gzuidhof/starboard-notebook/blob/3100df4602befa37c69dcae69041ce8a92fb360a/packages/starboard-notebook/src/components/helpers/localStorage.ts#L5-L10 - but apparently no longer.

**Comment by lfkpoa** [comment]
Thank you!
I'll take a look into it.
This issue was a problem because I was loading the observable plugin in a cell in the notebook, just like in the examples, but the warnings appear below  the cell that installs the plugin, as many times as there are observable cells.
@unhott told me in the Q&A how to pass additional plugins through the notebook properties and by doing that the warnings are not visible on the notebook, so this is not a big issue anymore.
I really like the reactivity between cells that observable gives us.
In my perspective, standard notebooks where you need to (re)run the cells, like jupyter, are good for people to reproduce and experiment with it.
But observable allows us to build notebooks easily for end users to interact with them. 
It can be used for storytelling, showing analysis and allowing the user to interact with the data through reactive controls and charts.
Arquero is an excellent javascript library to manipulate data in the browser and there are many great charting libraries too.
Recently I came across Echarts and I like it because it creates beautiful charts with interactivity by default.
Observable is the glue between fetching and manipulating the data and the visual components, without the need to create web applications.

**Comment by gzuidhof** [comment]
I agree one hundred percent. My intuition has been that the stuff you can do in the browser is simply not reactive by default, and that a notebook should not try to paper over that fact. It's great to build with plain browser capabilities fully exposed and being able to use any browser library (not just those that play nice with reactivity / needing to wrap those).

That said, I love the reactive programming paradigm. Starboard is very much adaptable enough to make it support entirely reactive cells (like it was shown in this plugin). My hobby wishlist has been to create a reactive runtime based on modern JS (e.g. cells would be ES modules, and any magic would be opt-in). 

The bottleneck is time and effort required :(. Building the product that is best (e.g. open source, hackable, somewhat standards-based) is sadly hard to make sustainable without taking external funding (which may turn it into a walled garden eventually).  Since then I started a b2b saas company doing something entirely unrelated to Starboard Notebook, which makes it hard to find the time to work on this a lot..
